workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DisconnectedState.cs

```diff
@@ -8,6 +10,7 @@ public DisconnectedState(UIManager manager, ConnectionStateMachine owner) : base
 
         public override void StartState()
         {
+            UnityObjectDestroyer.Destroy(Blackboard.ClientConnector.gameObject);
```

Comment:
This can be a normal `UnityEngine.Object.Destroy` if the gameObject exists for 1 frame or more, as the code is not run in editor mode.